frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -263,5 +266,42 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
                 break;
         }
     }
+
+    // The app cannot currently handle the back button action to allow users
+    // to move between screens back and forth. What happens is the app is "moved"
+    // to background as if "home" button were pressed.
+    // To avoid unexpected behaviour, we prompt users and force the app process
+    // to exit which helps with preserving phone's resources by shutting down
+    // all goroutines.
+    //
+    // Without forced app process exit, some goroutines may remain active even after
+    // the app resumption at which point new copies of goroutines are spun up.
+    // Note that this is different from users tapping on "home" button or switching
+    // to another app and then back, in which case no extra goroutines are created.
+    //
+    // A proper fix is to make the backend process run in a separate system thread.
+    // Until such solution is implemented, forced app exit seems most appropriate.
+    @Override
+    public void onBackPressed() {
+        new AlertDialog.Builder(MainActivity.this)
+            .setTitle("Close BitBoxApp")
+            .setMessage("Do you really want to exit?")
+            .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialog, int which) {
+                    // See the following for details about task stacks:
+                    // https://developer.android.com/guide/components/activities/tasks-and-back-stack
```

Comment:
It's a lot of text, and `moveTaskToBack` is not mentioned. A quick note why this is needed before killing the process would be helpful. It seems surprising, I am guessing it is not needed?

Reply:
You're right, it's essentially a noop today. Added it for future proof in case the app will have another activity in which case we'll most likely never remember to call moveTaskToBack before killing the process. If moveTaskToBack isn't called, the killProcess might actually kill a descendant proc so some remnants of the app would remain. At least, that's what I understood from reading the docs.

Will re-read again and add an explanation tonight.

Reply:
Turns out what I said earlier was incorrect. The `moveTaskToBack` isn't a noop. Without that call the system may attempt restarting the app because it can't tell whether it crashed or exited on purpose.

That's when in foreground. When in background, it doesn't really care. So, moving the process to the background is essential to avoid potential auto-restarts.